### ISSUE TITLE
Bun Install is Crashing

### ISSUE THREAD

**Issue by sagarsaud51** [oven-sh/bun#12356]
### How can we reproduce the crash?

Bun Add is Crashing:

```
bun add dbmate
```

### Relevant log output

```
Bun v1.1.18 (5a0b9352) Windows x64 (baseline)
Args: "C:\Program Files\nodejs\node_modules\bun\bin\bun.exe" "install"
Elapsed: 15ms | User: 0ms | Sys: 0ms
RSS: 43.35MB | Peak: 43.35MB | Commit: 90.59MB | Faults: 10764

panic(main thread): reached unreachable code
oh no: Bun has crashed. This indicates a bug in Bun, not your code.

To send a redacted crash report to Bun's team,
please file a GitHub issue using the link below:
```

### Stack Trace (bun.report)

Bun v1.1.18 ([`5a0b935`](<https://github.com/oven-sh/bun/tree/5a0b93523134ca0043ee7ae84f01003aa4b7ff69>)) on windows x86_64_baseline [InstallCommand]

**panic**: reached unreachable code

- [`Allocator.zig:98`](<https://github.com/oven-sh/bun/blob/5a0b93523134ca0043ee7ae84f01003aa4b7ff69/src/deps/zig/lib/std/mem/Allocator.zig#L98>): `rawFree`
- [`hash_map.zig:1333`](<https://github.com/oven-sh/bun/blob/5a0b93523134ca0043ee7ae84f01003aa4b7ff69/src/deps/zig/lib/std/hash_map.zig#L1333>): `getOrPutContextAdapted`
- [`install.zig:2760`](<https://github.com/oven-sh/bun/blob/5a0b93523134ca0043ee7ae84f01003aa4b7ff69/src/install/install.zig#L2760>): `getWithPath`
- [`lockfile.zig:4360`](<https://github.com/oven-sh/bun/blob/5a0b93523134ca0043ee7ae84f01003aa4b7ff69/src/install/lockfile.zig#L4360>): `processWorkspaceNamesArray`
- [`install.zig:8349`](<https://github.com/oven-sh/bun/blob/5a0b93523134ca0043ee7ae84f01003aa4b7ff69/src/install/install.zig#L8349>): `initWithCLI`
- [`cli.zig:1494`](<https://github.com/oven-sh/bun/blob/5a0b93523134ca0043ee7ae84f01003aa4b7ff69/src/cli.zig#L1494>): `start`
- [`cli.zig:62`](<https://github.com/oven-sh/bun/blob/5a0b93523134ca0043ee7ae84f01003aa4b7ff69/src/cli.zig#L62>): `start`
- [`main.zig:51`](<https://github.com/oven-sh/bun/blob/5a0b93523134ca0043ee7ae84f01003aa4b7ff69/src/main.zig#L51>): `main`
- [`exe_common.inl:78`](<https://github.com/oven-sh/bun/blob/5a0b93523134ca0043ee7ae84f01003aa4b7ff69/src/src/startup/exe_common.inl#L78>): `invoke_main`
- ??? at `0x1257d` in KERNEL32.DLL

**Comment by Jarred-Sumner** [comment]
@dylan-conway was this fixed?

**Comment by dylan-conway** [comment]
Fixed in Bun v1.1.39 with #15651
